test/e2e/l7networkpolicy_test.go

```diff
@@ -0,0 +1,233 @@
+	podSelector,
+	appliedToPodSelector map[string]string,
+	l4Protocol AntreaPolicyProtocol,
+	port int32,
+	l7Protocols []crdv1alpha1.L7Protocol) {
+	anpBuilder := &AntreaNetworkPolicySpecBuilder{}
+	anpBuilder = anpBuilder.SetName(data.testNamespace, name).SetPriority(priority)
+	if isIngress {
+		anpBuilder.AddIngress(l4Protocol,
+			&port,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+			l7Protocols,
+			nil,
+			nil,
+			podSelector,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+			[]ANPAppliedToSpec{{PodSelector: appliedToPodSelector}},
+			crdv1alpha1.RuleActionAllow,
+			"",
+			"")
+	} else {
+		anpBuilder.AddEgress(l4Protocol,
+			&port,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+			l7Protocols,
+			nil,
+			nil,
+			podSelector,
+			nil,
+			nil,
+			nil,
+			nil,
+			nil,
+			[]ANPAppliedToSpec{{PodSelector: appliedToPodSelector}},
+			crdv1alpha1.RuleActionAllow,
+			"",
+			"")
+	}
+
+	anp := anpBuilder.Get()
+	log.Debugf("creating ANP %v", anp.Name)
+	_, err := data.crdClient.CrdV1alpha1().NetworkPolicies(data.testNamespace).Create(context.TODO(), anp, metav1.CreateOptions{})
+	assert.NoError(t, err)
+}
+
+func testL7AntreaNetworkPolicyHTTP(t *testing.T, data *TestData) {
+	clientPodName := "test-l7-http-client-selected"
+	clientPodLabels := map[string]string{"test-l7-http-e2e": "client"}
+	cmd := []string{"bash", "-c", "sleep 3600"}
+
+	// Create a client Pod which will be selected by test L7 NetworkPolices.
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(clientPodLabels).Create(data))
+	if _, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", clientPodName, err)
+	}
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, clientPodName, data.testNamespace))
+
+	serverPodName := "test-l7-http-server"
+	serverPodLabels := map[string]string{"test-l7-http-e2e": "server"}
+	cmd = []string{"bash", "-c", "/agnhost netexec --http-port=8080"}
+	require.NoError(t, NewPodBuilder(serverPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(serverPodLabels).Create(data))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, serverPodName, data.testNamespace)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", serverPodName, err)
+	}
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, serverPodName, data.testNamespace))
+	var serverIPs []*net.IP
+	if podIPs.ipv4 != nil {
+		serverIPs = append(serverIPs, podIPs.ipv4)
+	}
+	if podIPs.ipv6 != nil {
+		serverIPs = append(serverIPs, podIPs.ipv6)
+	}
+
+	l7ProtocolPermitsPathHostname := []crdv1alpha1.L7Protocol{
+		{
+			HTTP: &crdv1alpha1.HTTPProtocol{
+				Method: "GET",
+				Path:   "/hostname",
+			},
+		},
+	}
+	l7ProtocolPermitsAnyPath := []crdv1alpha1.L7Protocol{
+		{
+			HTTP: &crdv1alpha1.HTTPProtocol{
+				Method: "GET",
+			},
+		},
+	}
```

Comment:
I don't get the point of this rule, it seems the result is same with or without it.